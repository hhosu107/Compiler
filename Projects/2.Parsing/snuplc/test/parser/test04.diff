64,65c64
<               neg <int>
<                 1 <int>
---
>               -1 <int>
