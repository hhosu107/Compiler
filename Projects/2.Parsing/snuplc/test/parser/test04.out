parsing 'test04.mod'...
successfully parsed.
  AST:
    CAstScope: 'test04'
      symbol table:
        [[
          [ *DIM(<ptr(4) to <NULL>>,<int>) --> <int>           ]
          [ *DOFS(<ptr(4) to <NULL>>) --> <int>           ]
          [ *ReadInt() --> <int>           ]
          [ *WriteChar(<char>) --> <NULL>           ]
          [ *WriteInt(<int>) --> <NULL>           ]
          [ *WriteLn() --> <NULL>           ]
          [ *WriteStr(<ptr(4) to <array  of <char>>>) --> <NULL>           ]
          [ @a        <int>           ]
          [ @b        <int>           ]
          [ @c        <int>           ]
        ]]
      statement list:
        := <int>
          [ @a        <int> ] <int>
          1 <int>
        := <int>
          [ @b        <int> ] <int>
          2 <int>
        := <int>
          [ @c        <int> ] <int>
          3 <int>
        := <int>
          [ @a        <int> ] <int>
          add <int>
            add <int>
              [ @a        <int> ] <int>
              [ @b        <int> ] <int>
            [ @c        <int> ] <int>
        := <int>
          [ @b        <int> ] <int>
          sub <int>
            sub <int>
              [ @a        <int> ] <int>
              [ @b        <int> ] <int>
            [ @c        <int> ] <int>
        := <int>
          [ @c        <int> ] <int>
          add <int>
            [ @a        <int> ] <int>
            mul <int>
              [ @b        <int> ] <int>
              [ @c        <int> ] <int>
        := <int>
          [ @a        <int> ] <int>
          neg <int>
            [ @a        <int> ] <int>
        := <int>
          [ @b        <int> ] <int>
          div <int>
            add <int>
              [ @a        <int> ] <int>
              [ @b        <int> ] <int>
            [ @c        <int> ] <int>
        := <int>
          [ @c        <int> ] <int>
          neg <int>
            add <int>
              neg <int>
                1 <int>
              [ @a        <int> ] <int>
      nested scopes:
        empty.
    


run the following command to convert the .dot file into a PDF:
  dot -Tpdf -otest04.mod.ast.pdf test04.mod.ast.dot


Done.
